components/Table/Table.js

```diff
@@ -95,12 +118,186 @@ export class Table {
+      menuItem.classList.add('menu-item');
+      menuItem.textContent = menuElement.label;
+
+      menuItem.addEventListener('click', () => {
+        const dataKey = cell.getAttribute('data-key');
+        switch (menuElement.key) {
+          case this.menuElMap.UNSORT:
+            if (this.callbacksMap.sortCallback) {
+              this.callbacksMap.sortCallback();
+            }
+            break;
+          case this.menuElMap.SORT_BY_ASC:
+            if (this.callbacksMap.sortCallback) {
+              this.isOrderAsc = true;
+              this.callbacksMap.sortCallback(dataKey, this.isOrderAsc);
+            }
+            break;
+          case this.menuElMap.SORT_BY_DESC:
+            if (this.callbacksMap.sortCallback) {
+              this.isOrderAsc = false;
+              this.callbacksMap.sortCallback(dataKey, this.isOrderAsc);
+            }
+            break;
+          case this.menuElMap.FILTER:
+            this.createFilterMenu();
+            break;
+          case this.menuElMap.HIDE:
+            cell.style.display = 'none';
+            const dataElements = this.tableElementsMap[dataKey];
+            if (dataElements) {
+              this.tableElementsMap[dataKey].forEach(
+                (item) => (item.style.display = 'none')
+              );
+            }
+            break;
+        }
+      });
+
+      return menuItem;
+    });
+    dropDownMenu.append(...this.menuDropDownItems);
+    menuBtn.append(dropDownMenu);
+
+    menuBtn.addEventListener('click', () => {
+      dropDownMenu.classList.toggle('show');
+    });
+  }
+
+  createFilterMenu() {
+    const filterMenuContainer = document.createElement('div');
+    filterMenuContainer.classList.add('filter-menu-container');
+
+    const filterLabel = document.createElement('div');
+    filterLabel.classList.add('filter-label');
+    filterLabel.textContent = 'Name';
+
+    const filterOperator = document.createElement('div');
+    filterOperator.classList.add('filter-operator');
+    filterOperator.textContent = 'equal';
+
+    const filterValue = document.createElement('input');
+    filterValue.classList.add('filter-value');
+    filterValue.placeholder = 'Filter value';
+
+    const btnClean = document.createElement('div');
+    btnClean.classList.add('btn-clean');
+    btnClean.textContent = 'x';
+
+    const btnExit = document.createElement('div');
+    btnExit.classList.add('btn-exit');
+    btnExit.textContent = 'x';
+
+    btnExit.addEventListener('click', () => {
+      filterMenuContainer.remove();
+    });
+
+    btnClean.addEventListener('click', () => {
+      filterValue.value = '';
+      if (this.callbacksMap.sortCallback) {
+        this.callbacksMap.sortCallback();
+      }
+    });
+
+    const filterFunction = (e) => {
+      let filter = e.target.value;
+      if (this.callbacksMap.filterCallback) {
+        this.callbacksMap.filterCallback(filter);
+      }
+    };
+
+    filterValue.addEventListener('input', filterFunction);
```

Comment:
Additionally, if you do not have `filterCallback` which sense to make event listener here?
Check if you have `filterCallback` and if so, set listener